### PR TITLE
Add missing typing support to make-fetch-happen.d.ts

### PR DESCRIPTION
**Pull request by michael-watson**
Add `cert`, `key`, `proxy` and `noProxy` options to `make-fetch-happen` typings to support internal security requirements around custom certs

### CODE REVIEW

File: gateway-js/src/make-fetch-happen.d.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ declare module 'make-fetch-happen' {
           randomize?: boolean;
         };
     onRetry?(): void;
+    cert?: string;
+    key?: string;
+    proxy?: string;
+    noProxy?: string;
```
Comment (trevor-scheer) [inline gateway-js/src/make-fetch-happen.d.ts:39]:
```suggestion
    noProxy?: string | string[];
```

File: gateway-js/src/make-fetch-happen.d.ts [hunk 2]
```diff
@@ -33,6 +33,10 @@ declare module 'make-fetch-happen' {
           randomize?: boolean;
         };
     onRetry?(): void;
+    cert?: string;
+    key?: string;
+    proxy?: string;
```
Comment (trevor-scheer) [inline gateway-js/src/make-fetch-happen.d.ts:38]:
```suggestion
    proxy?: string | URL;
```

We'll also need to add the proper import at the top:
`import { URL } from 'url';`

File: gateway-js/src/make-fetch-happen.d.ts [hunk 3]
```diff
@@ -33,6 +33,10 @@ declare module 'make-fetch-happen' {
           randomize?: boolean;
         };
     onRetry?(): void;
+    cert?: string;
+    key?: string;
```
Comment (trevor-scheer) [inline gateway-js/src/make-fetch-happen.d.ts:37]:
```suggestion
    cert?: SecureContextOptions['cert'];
    key?: SecureContextOptions['key'];
```
We'll also need to add the proper import at the top:
`import { SecureContextOptions } from 'tls';`

Comment (trevor-scheer) [reply gateway-js/src/make-fetch-happen.d.ts:37]: Ref: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/tls.d.ts#L563

File: gateway-js/src/make-fetch-happen.d.ts (hunk 1 above)
Comment (trevor-scheer) [reply gateway-js/src/make-fetch-happen.d.ts:39]: Ref: https://github.com/npm/make-fetch-happen#opts-no-proxy

File: gateway-js/src/make-fetch-happen.d.ts (hunk 2 above)
Comment (trevor-scheer) [reply gateway-js/src/make-fetch-happen.d.ts:38]: Ref: https://github.com/npm/make-fetch-happen#opts-proxy

**Review by trevor-scheer — CHANGES_REQUESTED** [review]
Left some suggestions. Noting that these are incomplete but definitely an improvement! These will be replaced / removed entirely when @abernix is able to land his `DefinitelyTyped` PR here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50339.

**Comment by trevor-scheer** [comment]
Please add a changelog entry, address comments and this will be good to go. Thanks @michael-watson!

**Comment by abernix** [comment]
Going to close this in favor of using the DefinitelyTyped-published types [which already include these missing properties](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d73fdfc3ff6aed8f47bd4092ada9623bcc1f4fe8/types/make-fetch-happen/index.d.ts).

Someone still needs to open the PR to change this repository to rely on that `@types/make-fetch-happen` package (and remove these baked-in types from this repo), but that's being tracked in https://github.com/apollographql/federation/issues/535.

Thanks for opening this originally! (If there are additional updates, the PRs should be opened to DefinitelyTyped.)

**Pull request closed without merging**